src/managed/OpenLiveWriter.SpellChecker/WinSpellingChecker.cs

```diff
@@ -0,0 +1,167 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OpenLiveWriter.SpellChecker
+{
+    public class WinSpellingChecker : ISpellingChecker, IDisposable
+    {
+        private PlatformSpellCheck.SpellChecker _speller;
+        private string _bcp47Code;
+
+        public bool IsInitialized
+        {
+            get
+            {
+                return _speller != null;
+            }
+        }
+
+        public event EventHandler WordAdded;
+        public event EventHandler WordIgnored;
+
+        public void AddToUserDictionary(string word)
+        {
+            CheckInitialized();
+            _speller.Add(word);
+
+            if (WordAdded == null)
+                return;
+            WordAdded(word, EventArgs.Empty);
+        }
+
+        public SpellCheckResult CheckWord(string word, out string otherWord, out int offset, out int length)
+        {
+            CheckInitialized();
+            otherWord = null;
+
+            if (string.IsNullOrEmpty(word))
+            {
+                offset = 0;
+                length = word.Length;
+                return SpellCheckResult.Correct;
+            }
+
+            PlatformSpellCheck.SpellingError spellerStatus = _speller.Check(word).FirstOrDefault();
+
+            if (spellerStatus == null)
+            {
+                offset = -1;
+                length = -1;
```

Comment:
I'd recommend setting `offset = 0;` and `length = word.length;` instead of -1. Or fixing up the handling in SpellCheckerForm.ContinueSpellCheck, because the `SpellCheckerForm.ContinueSpellCheck()` method will end up passing this length to [String.Substring](https://msdn.microsoft.com/en-us/library/aka44szs(v=vs.110).aspx) which will throw an ArgumentOutOfRangeException if its less than zero.